ext/bigdecimal/bigdecimal.c

```diff
@@ -6164,7 +6119,7 @@ VpDivd(Real *c, Real *r, Real *a, Real *b)
     word_c = c->MaxPrec;
     word_r = r->MaxPrec;
 
-    if (word_a >= word_r) goto space_error;
+    if (word_a >= word_r || word_b + word_c - 2 >= word_r) goto space_error;
```

Comment:
There are two more `goto space_error` in this function.
One is on [L6218](https://github.com/ruby/bigdecimal/blob/07facd4f2c14346ec72755df3eab57ec3a15c28f/ext/bigdecimal/bigdecimal.c#L6218)
```c
ind_r = ind_c + ind_b;
if (ind_r >= word_r) goto space_error;
```
And another in [L6253](https://github.com/ruby/bigdecimal/blob/07facd4f2c14346ec72755df3eab57ec3a15c28f/ext/bigdecimal/bigdecimal.c#L6253)
```c
ind_r = b->Prec + ind_c - 1; // L6242
ind_r = b->Prec + ind_c; // L6250 (ind_c + 1 < word_c is ensured in L6247)
if (ind_r >= word_r) goto space_error;
```
Both has the same requirement: `(word_b-1) + (word_c-1) < word_r`